test/common/utils_concurrency_limit.h

```diff
@@ -101,26 +107,242 @@ static int get_max_procs() {
     return maxProcs;
 }
 
-int get_start_affinity_process() {
+#if __TBB_USE_CGROUPS
+
+// Linux control groups support
+class cgroup_info {
```

Comment:
I think the implementation is good. Shall we copy it to the test for reference before merging the patch or maybe use some dumb version in tests that goes over the official paths without tricks to avoid parsing `/proc/self/mounts`?

Reply:
I think leaving more brute-force implementation in tests is good, because it will allow us to better verify whether our current more optimized implementation works correctly or not.

Reply:
Yeah. I have the same opinion.